ledger/partial/ptrie/partialTrie.go

```diff
@@ -160,7 +160,7 @@ func NewPSMT(
 		currentNode.path = path
 		// update node's hashvalue only for inclusion proofs (for others we assume default value)
 		if pr.Inclusion {
-			currentNode.hashValue = common.ComputeCompactValue(path, payload, currentNode.height)
+			common.ComputeCompactValue(&currentNode.hashValue, path, payload, currentNode.height)
```

Comment:
Isn't this an optimization that the compiler does ?

Reply:
In the previous version, a new slice is allocated in every call to the function to return the result (that happens inside `ComputeCompactValue`). Then `currentNode.hashValue` ignores its older value (thrown to GC), and points to the newer slice. 

In the new version, `currentNode.hashValue` is passed as an input so that the result is written directly in that slice, without allocating any new slice.